sycl/source/detail/helpers.hpp

```diff
@@ -81,6 +112,35 @@ template <typename iterator> class iterator_range {
   bool empty() const { return Size == 0; }
   decltype(auto) front() const { return *begin(); }
 
+  template <typename Container>
+  std::enable_if_t<
+      check_type_in_v<Container, std::vector<sycl_type>,
+                      std::queue<value_type *>, std::vector<value_type *>>,
```

Comment:
`std::vector<value_type *>` isn't used in this PR, but I have uses locally for my WIP patches related to `device_impl` raw ptrs/refs.